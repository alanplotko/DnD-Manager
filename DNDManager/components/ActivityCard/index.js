import React, { Component } from 'react';
import {
  Container,
  Content,
  Card,
  CardItem,
  Text,
  Body
} from 'native-base';
import { StyleSheet } from 'react-native';

export default class ActivityCard extends Component {
  render() {
    return (
      <Container>
        <Content>
          <Card>
            <CardItem header>
              {
                this.props.header &&
                <Text style={styles.header}>{this.props.header}</Text>
              }
            </CardItem>
            <CardItem>
              {
                this.props.body &&
                <Body>
                  <Text style={styles.body}>{this.props.body}</Text>
                </Body>
              }
            </CardItem>
            <CardItem footer>
              {
                this.props.footer &&
                <Text>{this.props.footer}</Text>
              }
            </CardItem>
         </Card>
        </Content>
      </Container>
    );
  }
}

const styles = StyleSheet.create({
  header: {
    fontWeight: 'bold',
    fontFamily: 'Roboto',
    fontSize: 18
  },
  body: {
    fontFamily: 'Roboto',
    fontSize: 16
  }
});
