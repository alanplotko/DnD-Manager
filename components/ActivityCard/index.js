import React from 'react';
import moment from 'moment';
import PropTypes from 'prop-types';
import { Card, CardItem, Text, Left, Body }
  from 'native-base';
import { StyleSheet, Image } from 'react-native';

const ActivityCard = props => (
  <Card style={{ marginLeft: 10, marginRight: 10, marginTop: 10 }}>
    <CardItem cardBody>
      <Left>
        <Image
          style={styles.thumbnail}
          resizeMode="cover"
          source={props.activity.thumbnail}
        />
        <Body style={{ paddingLeft: 10 }}>
          <Text style={styles.heading}>
            {props.activity.action}
          </Text>
          <Text style={styles.subheading}>
            {props.activity.extra}&nbsp;&bull;&nbsp;
            {moment(props.activity.timestamp).fromNow()}
          </Text>
        </Body>
      </Left>
    </CardItem>
  </Card>
);

const styles = StyleSheet.create({
  heading: {
    fontFamily: 'RobotoLight',
    fontSize: 18,
    color: '#000',
  },
  subheading: {
    fontFamily: 'Roboto',
    fontSize: 14,
    color: '#999',
  },
  thumbnail: {
    height: 72,
    width: 72,
    borderTopLeftRadius: 2,
    borderBottomLeftRadius: 2,
  },
});

ActivityCard.propTypes = {
  activity: PropTypes.shape({
    thumbnail: PropTypes.number,
    action: PropTypes.string,
    extra: PropTypes.string,
    timestamp: PropTypes.number,
  }),
};

ActivityCard.defaultProps = {
  activity: undefined,
};

export default ActivityCard;
