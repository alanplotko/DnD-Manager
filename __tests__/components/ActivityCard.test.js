import React from 'react';
import ActivityCard from 'FifthEditionManager/components/ActivityCard';

describe('Activity Card', () => {
  const activity = {
    thumbnail: 1,
    action: 'Created a new character!',
    extra: 'First Last',
    timestamp: Date.now(),
    icon: {
      name: 'add',
      color: '#fff',
    },
  };

  test('returns null without activity', () => {
    const wrapper = shallow(<ActivityCard />);
    expect(wrapper.type()).toEqual(null);
  });

  test('renders properly', () => {
    const wrapper = shallow(<ActivityCard activity={activity} uiTheme={DefaultTheme} />);
    expect(wrapper).toMatchSnapshot();
  });
});
