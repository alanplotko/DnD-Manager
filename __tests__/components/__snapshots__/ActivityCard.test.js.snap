// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Activity Card renders properly 1`] = `
<Styled(Card)
  style={
    Object {
      "marginBottom": 10,
    }
  }
>
  <Styled(CardItem)
    cardBody={true}
  >
    <Styled(Left)>
      <View>
        <Image
          blurRadius={10}
          resizeMode="cover"
          source={1}
          style={
            Object {
              "borderBottomLeftRadius": 2,
              "borderTopLeftRadius": 2,
              "height": 72,
              "width": 72,
            }
          }
        />
        <Icon
          color="#fff"
          name="add"
          size={48}
          style={
            Object {
              "left": 12,
              "position": "absolute",
              "top": 12,
            }
          }
        />
      </View>
      <Styled(Body)
        style={
          Object {
            "paddingLeft": 10,
          }
        }
      >
        <Styled(Text)
          style={
            Array [
              Object {
                "fontFamily": "RobotoLight",
                "fontSize": 18,
              },
              Object {
                "color": "#000000",
              },
            ]
          }
        >
          Created a new character!
        </Styled(Text)>
        <Styled(Text)
          style={
            Array [
              Object {
                "fontFamily": "Roboto",
                "fontSize": 14,
              },
              Object {
                "color": "#9e9e9e",
              },
            ]
          }
        >
          First Last
           • 
          a few seconds ago
        </Styled(Text)>
      </Styled(Body)>
    </Styled(Left)>
  </Styled(CardItem)>
</Styled(Card)>
`;
